dev-client/src/screens/SoilScreen/ColorScreen/utils/hooks.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useCallback} from 'react';
+import {launchImageLibraryAsync, MediaTypeOptions} from 'expo-image-picker';
+import {Photo} from 'terraso-mobile-client/screens/SoilScreen/ColorScreen/utils/image';
+
+export const usePickImage = (callback: (result: Photo) => void) =>
+  useCallback(async () => {
+    const response = await launchImageLibraryAsync({
+      mediaTypes: MediaTypeOptions.Images,
+      base64: true,
+    });
+    if (response.canceled) {
+      return;
+    }
+    const image = response.assets[0];
+    if (!image.base64) {
+      throw new Error('missing base64 from image picker!');
```

Comment:
question: what would case this to happen?

Reply:
assuming no bugs in the dependency, it would be because we fudged a config setting elsewhere and didn't instruct the image picker to return the base64 encoded data. i.e. the type of `image.base64` is `string | undefined`, with the actual value being one or the other depending on the arguments to `launchImagePickerAsync`. so here we're just making sure to fail fast if we do make a coding error and it's not coming back as `string` like we expected